android/src/com/mapswithme/maps/intent/Factory.java

```diff
@@ -804,13 +806,15 @@ else if (mLatFrom != null && mLonFrom != null)
       }
       else if (routerType > 0)
       {
-        RoutingController.get().prepare(true /* canUseMyPositionAsStart */,
+        MapObject startPoint = LocationHelper.INSTANCE.getMyPosition();
```

Comment:
Here and below, are we sure that LocationHelper.INSTANCE.getMyPosition() always returns not null?
Or it doesn't matter and RoutingController.prepare does it good?

Reply:
Both arguments are nullable:

```
  public void prepare(@Nullable MapObject startPoint, @Nullable MapObject endPoint)
  {
    prepare(startPoint, endPoint, false);
  }
```